subsys/bluetooth/mesh/shell/shell_time_cli.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+
+static void time_print(const struct shell *shell, int err, struct bt_mesh_time_status *rsp)
+{
+	if (!err) {
+		shell_print(shell,
+			    "Sec: %lld, subsec: %d, uncertainty: %lld, "
+			    "tai utc delta: %d, time zone offset: %d, is authority: %d",
+			    (uint64_t)rsp->tai.sec, rsp->tai.subsec, rsp->uncertainty,
+			    rsp->tai_utc_delta, rsp->time_zone_offset, rsp->is_authority);
+	}
+}
+
+static int cmd_time_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_TIME_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_time_cli *cli = mod->user_data;
+	struct bt_mesh_time_status rsp;
+
+	int err = bt_mesh_time_cli_time_get(cli, NULL, &rsp);
+
+	time_print(shell, err, &rsp);
+	return err;
+}
+
+static int cmd_time_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint64_t sec = (uint64_t)strtol(argv[1], NULL, 0);
+	uint8_t subsec = (uint8_t)strtol(argv[2], NULL, 0);
+	uint64_t uncertainty = (uint64_t)strtol(argv[3], NULL, 0);
+	int16_t tai_utc_delta = (int16_t)strtol(argv[4], NULL, 0);
+	int16_t time_zone_offset = (int16_t)strtol(argv[5], NULL, 0);
+	bool is_authority = (bool)strtol(argv[6], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_TIME_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_time_cli *cli = mod->user_data;
+	struct bt_mesh_time_status set = { .tai.sec = sec,
+					   .tai.subsec = subsec,
+					   .uncertainty = uncertainty,
+					   .tai_utc_delta = tai_utc_delta,
+					   .time_zone_offset = time_zone_offset,
+					   .is_authority = is_authority };
+
+	struct bt_mesh_time_status rsp;
+	int err = bt_mesh_time_cli_time_set(cli, NULL, &set, &rsp);
+
+	time_print(shell, err, &rsp);
+	return err;
+}
+
+static void zone_print(const struct shell *shell, int err, struct bt_mesh_time_zone_status *rsp)
+{
+	if (!err) {
+		shell_print(shell, "Current offset: %d, new offset: %d, timestamp: %lld",
+			    rsp->current_offset, rsp->time_zone_change.new_offset,
+			    rsp->time_zone_change.timestamp);
+	}
+}
+
+static int cmd_zone_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_TIME_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_time_cli *cli = mod->user_data;
+	struct bt_mesh_time_zone_status rsp;
+
+	int err = bt_mesh_time_cli_zone_get(cli, NULL, &rsp);
+
+	zone_print(shell, err, &rsp);
+	return err;
+}
+
+static int cmd_zone_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	int16_t new_offset = (int16_t)strtol(argv[1], NULL, 0);
+	uint64_t ts = (uint64_t)strtol(argv[2], NULL, 0);
```

Comment:
`strtol` returns `long int` to get `unsigned long int` you have to use `strtoul`